autoscale/src/main/java/com/sequenceiq/periscope/config/DatabaseConfig.java

```diff
@@ -92,7 +102,33 @@ private Properties jpaProperties() {
         properties.setProperty("hibernate.hbm2ddl.auto", hbm2ddlStrategy);
         properties.setProperty("hibernate.show_sql", "false");
         properties.setProperty("hibernate.dialect", "org.hibernate.dialect.PostgreSQLDialect");
+        properties.setProperty("hibernate.default_schema", StringUtils.isEmpty(dbSchemaName) ? "public" : dbSchemaName);
         return properties;
     }
 
+    private String createDbUrl() {
+        String url;
+        if (!StringUtils.isEmpty(dbSchemaName)) {
+            url = String.format("jdbc:postgresql://%s/%s?currentSchema=%s", databaseAddress, dbName, dbSchemaName);
+        } else {
+            url = String.format("jdbc:postgresql://%s/%s", databaseAddress, dbName);
+        }
+        return url;
+    }
+
+    private void createSchemaIfNeeded() {
+        if (!StringUtils.isEmpty(dbSchemaName)) {
+            try {
+                SimpleDriverDataSource ds = new SimpleDriverDataSource();
+                ds.setDriverClass(Driver.class);
+                ds.setUrl(String.format("jdbc:postgresql://%s/%s", databaseAddress, dbName));
+                Connection conn = ds.getConnection(dbUser, dbPassword);
+                Statement statement = conn.createStatement();
+                statement.execute("CREATE SCHEMA IF NOT EXISTS " + dbSchemaName);
+                conn.close();
+            } catch (Exception any) {
+                LOGGER.error(String.format("Cannot create the %s schema on %s in %s database", dbSchemaName, databaseAddress, dbName), any);
```

Comment:
In that case i guess a simple log message is not enough. Would be better to stop Cloudbreak as well and do do anything else till it is fixed.